storm-client/src/jvm/org/apache/storm/security/auth/kerberos/KerberosSaslTransportPlugin.java

```diff
@@ -119,7 +119,7 @@ private Login mkLogin() throws IOException {
     @Override
     public TTransport connect(TTransport transport, String serverHost, String asUser) throws IOException, TTransportException {
         WorkerToken token = WorkerTokenClientCallbackHandler.findWorkerTokenInSubject(type);
-        if (token != null && asUser != null) {
```

Comment:
The reason I had `asUser != null`, is because tokens done support user impersonation, but I got it wrong, and it should have been `asUser == null`.  But this should be fine.  If it becomes an issue we can add in support for asUser if needed